### PR TITLE
replace goKitLogger in config generator

### PR DESCRIPTION
**Pull request by perebaj**
## Description

replace goKitLogger in config generator

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Review by simonpasquier — APPROVED** [review]
thanks!

File: pkg/prometheus/test_utils.go [hunk 1]
```diff
@@ -60,6 +60,11 @@ func MakeExpectedReadinessProbe() *v1.Probe {
...
 
-func NewLogger() log.Logger {
-	return level.NewFilter(log.NewLogfmtLogger(os.Stdout), level.AllowWarn())
+func NewLogger() *slog.Logger {
+	l, _ := logging.NewLoggerSlog(logging.Config{
+		Level:  logging.LevelWarn,
+		Format: logging.FormatLogFmt,
+	})
+
+	return l
```
[the remark this answers is not in the document]
Comment (perebaj) [reply pkg/prometheus/test_utils.go:69]: @simonpasquier can you help me with this detail here?

[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/prometheus/test_utils.go:69]: I would panic if the function is only called in tests.

[the remark this answers is not in the document]
Comment (perebaj) [reply pkg/prometheus/test_utils.go:69]: adressing

[the remark this answers is not in the document]
Comment (perebaj) [reply pkg/prometheus/test_utils.go]: c183f52

**Review by ArthurSens — APPROVED** [review]
🥳

**Pull request merged**